### ISSUE TITLE
variable substitution occur only once when delegate_to using with serial

### ISSUE THREAD

**Issue by RanHamada** [ansible/ansible#9884]
##### Issue Type:

Bug Report
##### Ansible Version:

1.7.2
##### Environment:

CentOS 6.4
##### Summary:

Variable substitution occur only once when delegate_to using with serial.
##### Steps To Reproduce:

hosts

```
host1  copy_from=host9
host2  copy_from=host8
host3  copy_from=host7
```

playbook.yml

```

---
- hosts: all
  serial: 1
  tasks:
   - debug: msg={{ copy_from }}
     delegate_to: "{{ copy_from }}"
```
##### Expected Results:

```
TASK: [debug msg={{ copy_from }}] *********************************************
ok: [host3 -> host7] => {
    "msg": "host7"
}

TASK: [debug msg={{ copy_from }}] *********************************************
ok: [host2 -> host8] => {
    "msg": "host8"
}

TASK: [debug msg={{ copy_from }}] *********************************************
ok: [host1 -> host9] => {
    "msg": "host9"
}
```
##### Actual Results:

```
TASK: [debug msg={{ copy_from }}] *********************************************
ok: [host3 -> host9] => {
    "msg": "host9"
}

TASK: [debug msg={{ copy_from }}] *********************************************
ok: [host2 -> host9] => {
    "msg": "host8"
}

TASK: [debug msg={{ copy_from }}] *********************************************
ok: [host1 -> host9] => {
    "msg": "host7"
}
```
##### Workaround:

playbook.yml

```

---
- hosts: all
  serial: 1
  tasks:
   - debug: msg={{ copy_from }}
     delegate_to: "{{ item }}"
     with_items:
       - "{{ copy_from }}"
```

**Comment by sa2ajj** [comment]
Even with this workaround, `delegate_to: "{{ item }}"` fails with `Exception: host not found: {{ item }}` for me.

The correct host is shown in the log though.

**Comment by srgvg** [comment]
This is a duplicate of #8705. Should be solved in 1.8.x

**Comment by bcoca** [comment]
closing as dupe
